### DISCUSSION TITLE
Is there a RESTful API for Authentication in the non-self-hosted version?

### DISCUSSION THREAD

**Discussion by SnappierSoap318** [supabase/supabase#7633] [Questions] [answered]
im making an app in MIT Appinventor but it can only do REST calls, i cant make a quickly and run it as a webapp so is there a way to authenticate users using REST functions?

Will it be a security risk using rest apis instead of client libraries to make calls?

**Comment by GaryAustin1** [comment, marked as answer]
All the client libraries make rest API calls.  The gotrue(Auth) interface is REST.

This shows some of the endpoints (I don't know that it is a completely updated doc)  https://github.com/supabase/gotrue#endpoints

You could look at any of the client code like gotrue-js for ideas on the calls and what you have to do that the clients add on top of that.  Mainly you will have to deal with refreshing the jwt which gotrue-js handles.

**Reply by SnappierSoap318** [reply]
is the GoTrue based auth available on the supabase hosted service?

**Reply by GaryAustin1** [reply]
The link is to the gotrue version SB uses on their hosted service.

**Reply by SnappierSoap318** [reply]
`{base-url}/auth/v1/signup` with body as `{"email":"email@test.com","password":"secret"}` should work?

should secret be a hash?

**Reply by GaryAustin1** [reply]
That is a plain text password passed on and then encrypted in gotrue to the auth.users table.

**Reply by SnappierSoap318** [reply]
what is the URL that i should use to call the API?

**Reply by GaryAustin1** [reply]
I'm not sure I understand your issue at this point...  You showed the URL above for signup...
https://axxxxxxxxxxxxxxxxxxxxxu.supabase.co/auth/v1/signup
